pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,132 @@
+	secretLister   corev1listers.SecretLister
+	resourceSyncer resourcesynccontroller.ResourceSyncer
+}
+
+func NewCustomRouteCrontroller(
+	componentRouteNamespace string,
+	componentRouteName string,
+	destSecretName string,
+	destSecretNamespace string,
+	ingressInformer configinformers.IngressInformer,
+	ingressSetter configsetterv1.IngressInterface,
+	routeInformer routeinformer.RouteInformer,
+	secretInformer corev1informers.SecretInformer,
+	eventRecorder events.Recorder,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	operatorClient v1helpers.OperatorClient,
+) factory.Controller {
+	controller := &customRouteController{
+		destSecret: types.NamespacedName{
+			Namespace: destSecretNamespace,
+			Name:      destSecretName,
+		},
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		routeLister:    routeInformer.Lister(),
+		ingressLister:  ingressInformer.Lister(),
+		ingressSetter:  ingressSetter,
+		secretLister:   secretInformer.Lister(),
+		resourceSyncer: resourceSyncer,
+	}
+	return factory.New().
+		WithInformers(
+			ingressInformer.Informer(),
+			secretInformer.Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		WithSync(controller.sync).
+		ResyncEvery(1*time.Minute).
+		ToController("componentRouteSecretSyncController", eventRecorder)
+}
+
+func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	ingressConfig, err := c.ingressLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	// Update ingressConfig
+	route, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return err
+	}
+
+	ingressConfigCopy := ingressConfig.DeepCopy()
+	c.updateIngressConfigStatus(ctx, configv1.Hostname(route.Spec.Host), ingressConfigCopy)
+
+	// Copy source secret to destination
+	destination := resourcesynccontroller.ResourceLocation{Namespace: c.destSecret.Namespace, Name: c.destSecret.Name}
+	source := c.componentRouteSecretSource(ingressConfigCopy)
+
+	return c.resourceSyncer.SyncSecret(destination, source)
+}
+
+func (c *customRouteController) componentRouteSecretSource(ingress *configv1.Ingress) resourcesynccontroller.ResourceLocation {
+	// check if expected componentRoute exists
+	if secretName := common.GetCustomRouteServingCertKeySecret(ingress, c.componentRoute.Namespace, c.componentRoute.Name); secretName != "" {
+		if _, err := c.secretLister.Secrets("openshift-config").Get(secretName); err == nil {
+			return resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: secretName}
+		}
+	}
+	return resourcesynccontroller.ResourceLocation{}
+}
+
+func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, currentHostname configv1.Hostname, ingress *configv1.Ingress) error {
+	desiredComponentRoute := configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
```

Comment:
needs conditions

Reply:
Given that we are going to move the `handleRoute` method from the `metadataController` I can surface the conditions it generates here.

Reply:
Added a few basic conditions, still need to add more advanced tests to check service availability.

Reply:
Conditions should be good to go now.